tests/Feature/EditPageTest.php

```diff
@@ -178,6 +192,184 @@ public function itOnlyShowsAFoodMattersInputsToFoodMattersUsers()
         }
     }
 
+    /** @test */
+    public function itShowsDiaryAndChartCheckedCorrectlyAsStoredInDatabase()
+    {
+        // Create a User
+        $fmuser = factory(User::class)->create([
+            'name' => 'test fm user',
+            'email' => 'testfmuser@example.com',
+            'password' => bcrypt('test_fmuser_pass'),
+            'centre_id' => $this->centre->id,
+            'role' => 'foodmatters_user',
+        ]);
+
+        $now = Carbon::now();
+
+        // make no docs registration
+        $regs[] = factory(Registration::class)->create([
+            'centre_id' => $this->centre->id,
+            'fm_diary_on' => null,
+            'fm_chart_on' => null,
+        ]);
+
+        // make chart only registration
+        $regs[] = factory(Registration::class)->create([
+            'centre_id' => $this->centre->id,
+            'fm_diary_on' => $now,
+            'fm_chart_on' => null,
+        ]);
+
+        // make diary only registration
+        $regs[] = factory(Registration::class)->create([
+            'centre_id' => $this->centre->id,
+            'fm_diary_on' => null,
+            'fm_chart_on' => $now,
+        ]);
+
+        // make both docs registration
+        $regs[] = factory(Registration::class)->create([
+            'centre_id' => $this->centre->id,
+            'fm_diary_on' => $now,
+            'fm_chart_on' => $now,
+        ]);
+
+        foreach ($regs as $reg) {
```

Comment:
Since you are creating each reg - would it more clear to just name them each and visit the page. See what you expect. You can keep the reg[] array or even just make 4 $reg_both, $reg_chart $reg_diary

`visit(URL::route('service.registration.edit', 1)
->seeInDatabase('registrations', ['id' => 1, 'fma_chart_on' => $now]);
$this->seeElement('input[name="fm_chart"][checked]');`

This is fine as is - just might be less confusing to spell it out. And maybe a better test if you are not fetching the value of e.g. $reg->fm_chart_on to determine what you see.